java/src/jmri/server/json/util/JsonUtilSocketService.java

```diff
@@ -55,12 +55,66 @@ public void onMessage(String type, JsonNode data, JsonRequest request) throws IO
             case JSON.RAILROAD:
                 this.onRailroadNameMessage(type, data, request);
                 break;
+            case JSON.SESSION_LOGIN:
+                this.onSessionLoginMessage(type, data, request);
+                break;
+            case JSON.SESSION_LOGOUT:
+                this.onSessionLogoutMessage(type, data, request);
+                break;
             default:
                 this.connection.sendMessage(this.service.doPost(type, name, data, request), request.id);
                 break;
         }
     }
 
+    /**
+     * Process an incoming POST login message
+     *
+     * Extract username, password
+     * Check against authentication backend
+     *
+     * On success, send a response containing a valid token
+     * On Failure, send an exception message
+     *
+     * @param type Message type
+     * @param data JSON payload
+     * @param request The original request as received
+     * @throws IOException
+     * @throws JmriException
+     * @throws JsonException
+     */
+    private void onSessionLoginMessage(String type, JsonNode data, JsonRequest request) throws IOException, JmriException, JsonException {
+        String username = data.path("username").asText();
+        if (request.method.equals(JSON.POST)) {
+            log.debug("Processing login {} from socket service", username);
+            JsonNode resultNode = this.connection.sendMessage(this.service.doPost(type, username, data, request), request.id);
+        }
+//        this.connection.sendMessage(this.service.doGet(type, name, data, request), request.id);
+    }
+
+    /**
+     * Process an incoming POST logout message
+     *
+     * Extract credential
+     * Check against authentication backend
+     *
+     * On success, invalidate token. Send invalidated token.
+     * On Failure, send an exception message.
+     *
+     * @param type
+     * @param data
+     * @param request
+     * @throws IOException
+     * @throws JmriException
+     * @throws JsonException
+     */
+    private void onSessionLogoutMessage(String type, JsonNode data, JsonRequest request) throws IOException, JmriException, JsonException {
+        if (request.method.equals(JSON.POST)) {
+            // Do we really want the token logged?
+            JsonNode resultNode = this.connection.sendMessage(this.service.doPost(type, 'REDACTED', data, request), request.id);
```

Comment:
What do you mean with this comment? Does the method sendMessage() logs the permission sessionId?

Reply:
Also, you have a syntax error on line 114. `'REDACTED'` should be `"REDACTED"`.

Reply:
The doPost method does.

`        
log.debug("doPost(type='{}', name='{}', data='{}')", type, name, data);
`

We can move the log line further down into the switch statement to prevent that.

Reply:
I think we can leave it as is. The risk that someone would use the log to hack JMRI is low.